samcli/commands/_utils/custom_options/hook_name_option.py

```diff
@@ -58,7 +59,13 @@ def handle_parse_result(self, ctx, opts, args):
         if not _check_experimental_flag(hook_name, command_name, opts, ctx.default_map):
             return super().handle_parse_result(ctx, opts, args)
 
-        self._call_prepare_hook(iac_hook_wrapper, opts)
+        try:
+            self._call_prepare_hook(iac_hook_wrapper, opts)
+        except Exception as ex:
```

Comment:
a note for future refactor - seems like we will need to apply the same pattern in all click option callbacks to catch and handle exceptions. Can we create a decorator function to do this?